Cython/Compiler/Symtab.py

```diff
@@ -2705,12 +2706,13 @@ def declare_inherited_cpp_attributes(self, base_class):
                                            modifiers=base_entry.func_modifiers,
                                            utility_code=base_entry.utility_code)
             entry.is_inherited = 1
-        for base_entry in base_scope.type_entries:
+        for base_entry in base_scope.type_entries + base_scope.inherited_type_entries:
             if base_entry.name not in base_templates:
                 entry = self.declare_type(base_entry.name, base_entry.type,
                                           base_entry.pos, base_entry.cname,
-                                          base_entry.visibility)
+                                          base_entry.visibility, defining=0)
                 entry.is_inherited = 1
+                self.inherited_type_entries.append(entry)
```

Comment:
Here we are in the AST type declaration part.

The goal here is to make inherited types accessible for derived classes. Let's take an example.
I define a parent cppclass ClassA, and inside ClassA, I define a type MyNestedType. This is a type nested inside ClassA.
Now, I make a class ClassB which inherits from ClassA.

The goal of the patch in Symtab is to be able to write in cython code something like cdef ClassB.MyNestedType my_var.

I'm not at all a fluent reader of the C++ norms, so I don't know if that "nested type transitivity" is an actual feature of C++, I just saw that implementation-wise, g++ allows that, so I made that happen into the Cython compiler.

It looks like the example I just gave has no unit test. I'm gonna make one, hopefully soon.

Reply:
The behaviour sounds reasonable - it's consistent with what both Python and C++ does (although C++ behaviour depends a bit on the visibility).

On a quick read I'm not convinced we need the `self.inherited_type_entries`. Since you redeclare them, don't they end up in `type_entries`? (Or am I wrong here?)

Reply:
Oh you're right, I was quite blindly mimicking what is done for variables, where we use `declare()` for inherited variables and not `declare_var`. `self.inherited_type_entries` is indeed useless, that's on my todo-list for the next update of this PR.
Thanks for catching that mistake !